### DISCUSSION TITLE
Welcome to CircuitVerse Discussions!

### DISCUSSION THREAD

**Discussion by tachyons** [CircuitVerse/CircuitVerse#2702] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Kartikay304** [comment]
Hello everyone, I am kartikay kala. I am pursuing B.tech in CSE, and currently, I am in the 3rd semester. I have joined CircuitVerse in August 2021. The whole community at CircuitVerse has been super welcoming from the first day I started contributing. I learned a lot since I have joined CV got to learn a new language Ruby, a lot of awesome gems & best software practices in general.
I plan to continue contributing to CircuitVerse. So that I learn more, improve my coding and soft skills, and improve the end-user experience of users using cv.

**Reply by OsirisMedici** [reply]
Kartikay, can you tell me more about Circuit verse? You are contributing to CV for a time so For a beginner like me, you are the perfect match.
contactabhishek249@gmail.com

**Reply by swarnshekhar** [reply]
Hello..,

**Comment by AtharvaShobhane** [comment]
Hello everyone I am Atharva Shobhane , currently pursuing my BTech in Computer Engineering , I want to contribute to CV , and want to explore open source , can anyone help me with that

**Comment by lacthon** [comment]
Greetings of the day 
To contributors ,mentors,member of organization
I am Dipanshu joshi a computer science undergraduate.I have just entered in my 2nd year at Shriram Institute of Management and technology.I am new to open source contributions but i am well aware of c,c++,html and python(learning).I would love to contribute in  your organisation in gsoc 2023 but I am confused how to join or start.Will you plz help me to how to start  and I can connect as a contributor to your organisation in 2023.

 Hoping to hear from you soon

**Comment by lacthon** [comment]
contact : dipanshujoshi890@gmail.com

**Comment by AnmolChauhan123** [comment]
I am Anmol Chauhan, a computer science undergraduate. I have just completed my second year. I am really new to open source contributions but I am well aware of Python, C++, Javascript, node.js and many other libraries of python like OpenCV and MediaPipe. I would love to contribute to your organization but could you please tell me how to get started?
Hoping to hear from you soon.
Regards 
Anmol Chauhan
Contact - anmol345chauhan@gmail.com

**Comment by amankumar11** [comment]
I am Aman Kumar a third year student in Thapar Institute of Engineering and Technology pursuin Bachelors in Electronics and Communication , i am familiar with HTML, CSS, JS, MERN, Redux, Python, Flask, C++, C, Solidity, MySQL and MongoDB. As i am from ECE background i am familiar with logics gates and circuits.
I would like to contribute in CV can you pls guide me how to start with it.
Hoping to hear from you soon.
Regards
Aman Kumar
Contact - aman.kumar11112001@gmail.com

**Comment by DarkPheonix12** [comment]
I am Aman Gaur. I am currently in my 4th semester in Vellore Institute of Technology perusing my Bachelors in ECE , I have experience in Web Dev (HTML, CSS, JS, Node.js,React.js, Postman, MySQL, MongoDB), Blockchain Development (Solidity language and Truffle suit, Ganache suit) and DSA. I have worked on different projects with my friends while in hackathons as well as in my college. I really find CIRCUITVERSE interesting and I am really excited to contribute to this community. Please guide me in that direction.
Thank you.
Aman Gaur.
Email-aman.gaur5505@gmail.com

**Comment by Malavi1** [comment]
Hope you all doing well! Thank you CIRCUIT VERSE for the amazing platform of digital learning.So,my name is MalaviPande pursuing B.Tech electronics as a major I got in touch with open source from last 3 months and I really want to be a part of this organization by contributing and am quite familiar with python,HTML,CSS and JavaScript if the upcoming projects other than my tech stack ready to learn. My final aim is to collaborate contribute and network !!!

ThankYou !!

**Comment by DarkPheonix12** [comment]
Are you texting me because you want to be contributor in GSOC??

**Reply by amankumar11** [reply]
Yes sir, I want to be a contributor in GSOC.

**Reply by Malavi1** [reply]
Guys what was happening here ? could you please tell a bit !!

**Comment by volfiros** [comment]
Hello everyone,
This is Sai Rithvik Padma, a Computer Science undergrad at BITS Pilani. I recently started doing open-source contributions. I bumped across this organisation and I am looking forward to contribute to CircuitVerse. Can anyone suggest some good first issues to start contributing to the organization? 
I am a frontend developer with tech stack - HTML, CSS, Bootstrap, Javascript, jQuery, React.js, Node.js, python, c and java. I am always open to learning new technologies to become better in development.
My GitHub profile link: https://github.com/Rithvik-padma
Hoping to hear back from you soon!
Thanking You,
Sai Rithvik Padma

**Comment by Malavi1** [comment]
Hi @Rithvik-padma 
Get familiar with the contribution guidelines first and DO join the slack channel here by clicking this [Slack](https://circuitverse.org/slack). Introduce yourself and DO ask questions if you get stuck.
You can go through the other repositories by CircuitVerse as well and find the Issues with Good First Issues Label.
If that issue is not being worked upon by anyone, you can comment on it stating that you're working, and can raise a PR when done with the changes.
Try to set up the project using [SETUP.md](https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md) guidelines.

**Comment by swarnshekhar** [comment]
> Hello everyone, I am swarn shekhar, currently persuing my btech(IT-1st sem) in galgotia college of engineering.
Till date I have gathered skills such as-java,DSA,web development (html,css, javascript).
I am interested to contribute here...
I will be glad if bhaiyas(seniors)here will guide me in a direction to contribute and learning things.

**Comment by v-a14** [comment]
Hello Everyone, I am beginner to open source. I want to contribute to circuit Verse. 
Thanks.

**Reply by swarnshekhar** [reply]
I recently thought about an idea to create a discord server for beginners..
And we will grow upon it..

How you think about this...?

**Reply by tachyons** [reply]
@swarnshekhar We already have a slack workspace.

**Comment by Pratikdate** [comment]
Hi!
Myself, Date Pratik , Perceive CS Engineering at Pune university.
Techstacks i have worked on and comfortable with are, Python, Django, MySQL, HTML, CSS, JavaScript .
I am excited to contribute to the organization, and interested in learning new tech stacks required for the projects, in the process.

**Comment by Raoraya** [comment]
hello everyone this is Aman, have done my bsc degree in computer science. also learned some industry-required tech stacks like MERN stack and subsequent web dev tech like nextjs, PostgreSQL, GraphQL, Mono Repo, TRPC, Docker, etc. I'm looking forward to contributing to CircuitVerse

**Comment by Ayushhariwal** [comment]
Hi , i am Ayush ,  currently in my 2nd year of computer science engineering and also new to open source collaborations , but i aquire a good knowledge of  web development with MERN stack , I would like to contribute to your organisation but could you please tell me how to get started?

**Comment by Prajyot70** [comment]
Hello Everyone,
I'm Prajyot,
A 2nd year B.tech degree student curious and driven towards open source. Having some experience with github and open-source through past contributions, here to gain more.

**Comment by JamesCartar** [comment]
Hi, I am Hein Htet, a second-year Computer Science student. 

I got to know CircuitVerse while I was studying the Computer Systems curriculum where we have to learn about Circuits and logic gates. The tool that my University provided us with for building circuits with logic gates and doing our experimentations, was a bit complicated to use. So, I started searching for a good and simple platform to build circuits and I found CircuitVerse. Since then, I have been using CircuitVerse for my personal and educational purposes.

I am planning to apply for the Google Summer of Code (GSoC) this year with CircuitVerse. I believe that CircuitVerse is an excellent platform for both beginners and intermediate learners like me who are delving into the world of circuits and logic gates. 

I have observed some areas in the platform that could be improved, which would enhance the learning experience for everyone. My goal is to contribute to this improvement and make CircuitVerse even better for all of us.

**Comment by Pulkit0729** [comment]
Hi everyone, I am Pulkit, a software enginner from IIT Delhi. i was trying to join the slack channel through the link on website, but it is showing invalid. Please can someone post the new link.
Thanks

**Reply by DhairyaMajmudar** [reply]
Hi, @Pulkit0729 
Here is the link [Slack](https://join.slack.com/t/circuitverse-team/shared_invite/zt-2axe9h7hy-GwM~nL7pq1Jh1hSZYB~n1A) channel

**Reply by Pulkit0729** [reply]
Thanks 😊 @DhairyaMajmudar

**Comment by subhamkumarr** [comment]
Hi everyone, I am Subham,  IIT Dhanbad. I look forward to contributing to Circuitverse and have an impact. Thanks

**Comment by Nj221102** [comment]
Hi everyone , My name is nitish Jha , i am a computer science undergrad from india, I have keen interest in ruby and rails , and also have experience working with ruby stack when i was contributing to scarpe in past,  and with them i have decent skill in javascript , html ,css as well, so contributing to this project seems perfect for me so i am really excited and looking forward to contributing to Circuitverse, can someone suggest what will be the best way to get familiar with the codebase.

**Comment by Sanil-Surve** [comment]
Hii, I am Sanil Surve. Looking for to contribute in your organisation.  I was following Circuit Verse from 2022 and now I have seen that there is lot of scope in your organisation to work and get value addition in your Professional life and vice versa. so I want to become contributor in  your organisation for GSoC 2024. Do you interested in looking up at me? or not @Sanil-Surve

**Comment by radagon12** [comment]
Hi I am Anogh Dalal. I am interested to contribute to CircuitVerse, to learn how to basically analyse and understand some inherent code complexities while also adding some value to it. I have one of my friends already contributing at CircuitVerse and their experience at CircuitVerse enabled me to join the open source journey as well

Just one thing I am not finding the slack channel, can anyone share the link?

**Reply by subhamkumarr** [reply]
Hello 👋! Here you go. Feel free to ask all your questions in Slack.
https://circuitverse-team.slack.com/archives/CQFAC5L7P

**Comment by cybersteve547** [comment]
can someone help me? i cant figure out how to select multiple objects.

**Comment by salmoneatenbybear** [comment]
Does anyone think moving from bootstrap to tailwind would be a good decision given its customisability through low level utility classes and the enhanced performance of the site as a whole because of PurgeCSS which removes unused stuff on production

@tachyons @tanmoysrt

**Comment by salmoneatenbybear** [comment]
# Proposal: Community Library for Circuit Components

I’d like to propose a feature for CircuitVerse: a **Community Library of Circuit Components**, where users can share and reuse circuit designs to foster collaboration and learning.

## Full Vision:
The idea is to create a comprehensive library within CircuitVerse itself, where users can upload reusable circuit components with metadata like name, description, and tags. These components would be stored on the platform’s backend, with features for uploading, browsing, and importing directly into designs. A search feature would allow users to easily find components by name or tags, and attribution would be embedded into downloaded components to credit the original creator. To ensure quality, a moderation system and user-driven ratings could help maintain the library’s reliability and usefulness.

## Initial Approach:
To keep things simple as a starting point, users could upload basic circuit components in a lightweight format (e.g., JSON) with minimal metadata like name, description, and uploader. These components can be stored on the server and presented in a simple interface within the circuit editor. A basic list view or search bar can help users explore and integrate components into their projects. Attribution can be automatically included in the metadata of imported components. This approach ensures that the system remains simple while enabling the key functionality of sharing and reusing components effectively.

---

I’d love to hear everyone’s thoughts on this idea. How can we improve or expand this feature to make it more impactful for CircuitVerse users?


@tachyons

**Reply by ThatDeparted2061** [reply]
Some quick inclusions in this idea would be including the featured examples, editor picks, recent circuits with their proper names of their circuits in the library section.
This step will help us in the future development of CircuitVerse and will boost is usability and help users learn better with quick accessibility  to the important circuits.
![Screenshot 2024-11-26 183931](https://github.com/user-attachments/assets/5b732ef6-6510-4375-8ead-1f06a83e42a5)
![Screenshot 2024-11-26 184003](https://github.com/user-attachments/assets/5a39ed66-4153-4d5b-8821-6040e3052f2f)
![image](https://github.com/user-attachments/assets/ecdb5c06-6783-41c3-b076-db87d2660237)

**Reply by AdityaP700** [reply]
With a growing number of active users, this feature will become essential. Additionally, implementing hierarchical categories (e.g., basic gates, arithmetic units, memory elements, etc.) would be highly beneficial. These categories can be further enriched by assigning difficulty levels such as Beginner/Easy, Moderate, and Advanced. Moreover, linking educational resources to each component will provide users with valuable learning opportunities, helping them to expand their knowledge and skills effectively.

**Comment by trouteatenbybear** [comment]
Hiiii

**Comment by Rahul-Aitla** [comment]
hii

**Comment by IITI-tushar** [comment]
Hi everyone! 👋

I'm Tushar Saxena, currently 2nd year student at IIT Indore. I have a strong interest in open-source contributions and am excited to be part of the CircuitVerse community to help enhance this amazing platform.

I’m eager to work on new features, fix bugs, and collaborate with the community to make CircuitVerse even more impactful and user-friendly. Looking forward to learning, growing, and contributing to this fantastic platform! 🚀

It would be much appreciated if you could guide me with resources or tips to get started, especially on understanding the codebase and best practices for contributions.

@tachyons

**Comment by nidhaahmed** [comment]
Hello, 

I am Nidha Ahmed Mohammad, a junior undergraduate student majoring in Artificial Intelligence and Data Science. I’m also excited to share that I’ll be joining Google as an SWE Intern next summer.

I have experience with JavaScript, HTML5, Bootstrap, PostgreSQL, and machine learning. Although I’m new to Rails, Vue, and HTML5 Canvas, I am excited to learn and upskill myself to work effectively with your tech stack. If you could share any beginner-friendly resources or guidance for getting started with these technologies and your codebase, it would be incredibly helpful.

A little about me: I’ve worked on projects like a movie recommendation system, a geospatial analysis tool for public health, and open-source contributions like designing real-time heatmaps for SoundScape. My passion for combining education and technology aligns closely with CircuitVerse.org’s mission, and I’m eager to grow while making meaningful contributions.

It would mean a lot to receive guidance on how I can begin contributing to your community. I’m genuinely excited about the opportunity to support CircuitVerse.org’s impactful work.

Thank you...

**Comment by maheshcodes12** [comment]
@tachyons Hey!! I am new here. I'm eager to join the CircuitVerse community and willing to contribute here best as I can. I tried to join Slack workspace but the invitation links are expired everywhere. Can you help me with this, please?
![slackinvalid](https://github.com/user-attachments/assets/4476f95f-453f-4558-9d21-1c1c1e092e77)

**Reply by salmoneatenbybear** [reply]
> @tachyons Hey!! I am new here. I'm eager to join the CircuitVerse community and willing to contribute here best as I can. I tried to join Slack workspace but the invitation links are expired everywhere. Can you help me with this, please?
> ![slackinvalid](https://github.com/user-attachments/assets/4476f95f-453f-4558-9d21-1c1c1e092e77)
> 
> 

https://join.slack.com/t/circuitverse-team/shared_invite/zt-2yo74pd8x-3uhWIZzNpzJwHcbU7oLL3A

**Comment by Magnolia2423** [comment]
Hello !
I am Magnolia, a B.Tech Computer Science 1st-year student from MIT college, U.P. , India.

I am interested in contributing to CircuitVerse and GSoC 2026.
I have been trying to go through the code and learn about logic circuits.
I am looking forward to learning from this wonderful community!
Any advice for a beginner like me would be great.
Thank you!
Best regards,
Magnolia

**Comment by nusRying** [comment]
`Hi CircuitVerse team and mentors,

I'm planning to apply for **GSoC 2026 Project 3: FSM to Circuit Synthesizer** and would appreciate early feedback on my scope and technical approach.

**Prepared documentation:**
- **One-page design:** [GoSC/06-One-Page-Design.md](https://github.com/CircuitVerse/CircuitVerse/blob/main/GoSC/06-One-Page-Design.md) (schema, Moore/Mealy conversion flow, output representation)
- **Timeline & milestones:** [GoSC/02-Timeline-and-Milestones.md](https://github.com/CircuitVerse/CircuitVerse/blob/main/GoSC/02-Timeline-and-Milestones.md)
- **Execution plan:** [GoSC/05-This-Week-Lock-In-Plan.md](https://github.com/CircuitVerse/CircuitVerse/blob/main/GoSC/05-This-Week-Lock-In-Plan.md)

**Proposed MVP (175 hours):**
- Structured/tabular FSM input workflow
- Deterministic validation and binary state encoding
- Boolean equation generation for Moore and Mealy machines
- Synthesis to editable CircuitVerse-compatible sequential circuits

**Scope questions for feedback:**
1. Is this MVP realistic for a 175-hour timeline?
2. Would Moore-first then Mealy sequencing reduce implementation risk?
3. Which simulator integration boundary should be prioritized first for reliability and review efficiency?

**Demonstrating execution quality:** I'm preparing 1-2 contribution PRs this week (bugfix/test-focused) to show implementation rigor before proposal review.
[02-Timeline-and-Milestones.md](https://github.com/user-attachments/files/25737307/02-Timeline-and-Milestones.md)
[05-This-Week-Lock-In-Plan.md](https://github.com/user-attachments/files/25737309/05-This-Week-Lock-In-Plan.md)
[06-One-Page-Design.md](https://github.com/user-attachments/files/25737311/06-One-Page-Design.md)


Looking forward to your guidance. Thank you!`

**Comment by vj-codes** [comment]
Hello! The Slack link provided on gsoc page and website is no longer active
Please suggest a way to join the communication channel!

**Reply by nusRying** [reply]
sure let me look for it

**Comment by programmeo** [comment]
Hello mentors,

**I’m Yash** and I’m interested in contributing to CircuitVerse.

I mainly work with the MERN stack and enjoy building interactive web applications. I came across CircuitVerse while looking for open-source projects where I could contribute and possibly **prepare for GSoC 2026**.

I’ve started reading through the repository and documentation. **If there are any beginner issues or areas where new contributors can help**, I’d love to get involved.

Thanks!

**Comment by Thesmoothengineer** [comment]
Hey! I'm a new contributor looking to get started with CircuitVerse for GSoC 2026. The Slack invite link seems to be expired, could someone share an updated one? Thanks!

**Reply by tachyons** [reply]
Please use https://join.slack.com/t/circuitverse-team/shared_invite/zt-3spixgmk0-v601OQMWVEIH8nKseQpzXw while we are working on it

**Comment by rizwan749** [comment]
Rizwan (GSoC 2026 Aspirant)

Hi everyone! I'm Rizwan, a 4th-semester Software Engineering student and a React developer. I've been exploring the platform and I'm really excited to start my open-source journey by contributing to CircuitVerse for GSoC 2026. Looking forward to learning from this amazing community and helping out!

**Comment by krishnavasnani07** [comment]
Hi, I’ve submitted my GSoC 2026 proposal for the FSM to Circuit Synthesizer project. 
I’ve also made some contributions to CircuitVerse.

I would really appreciate any feedback to improve my proposal further.

Thank you!
<img width="1536" height="1024" alt="FSM_architecture" src="https://github.com/user-attachments/assets/a45e8db3-eae5-4daf-a127-8b6531b79017" />

**Comment by maltheshhh** [comment]
Hey CircuitVerse team! 👋

I’m a new contributor to CircuitVerse. I actually used CircuitVerse during my earlier school studies to learn and experiment with digital circuits, so it’s pretty cool to now contribute to the project itself. 😄

I made my very first PR today, and I’m really excited to start contributing more and learn along the way. Looking forward to being part of the CircuitVerse community! 🚀

Thanks for building such a great open-source project! ❤️
